microsoft-edge/devtools-guide-chromium/evaluate-performance/index.md

```diff
@@ -183,22 +176,68 @@ This article gives you a lot to learn. But now you have a solid foundation in th
+Compare this snippet of JavaScript from the unoptimized version of the app:
+
+```javascript
+var pos = m.classList.contains("down")
+  ? m.offsetTop + distance
+  : m.offsetTop - distance;
+if (pos < 0) pos = 0;
+if (pos > maxHeight) pos = maxHeight;
+m.style.top = pos + "px";
+if (m.offsetTop === 0) {
+  m.classList.remove("up");
+  m.classList.add("down");
+}
+if (m.offsetTop === maxHeight) {
+  m.classList.remove("down");
+  m.classList.add("up");
+}
+```
+
+The above code snippet runs on every frame of the browser rendering loop, for each blue icon on the page.  The `m` variable references a single blue icon.
+
+In this unoptimized version, we create a `pos` variable that's set to the current position of the icon, to which we add some distance.  The current position of the icon is read by using `m.offsetTop`.
+
+After making sure that the icon is still within the bounds of the page, we set its new position by using `m.style.top`, which sets inline styles on the element.
 
+Finally, we read `m.offsetTop` again, to adjust the direction of the icon.
+
+The optimized code uses a different sequence of actions to do less work. Here is the same snippet of JavaScript from the optimized version of the app: 
+
+```javascript
+var pos = parseInt(m.style.top.slice(0, m.style.top.indexOf("px")));
+m.classList.contains("down") ? (pos += distance) : (pos -= distance);
+if (pos < 0) pos = 0;
+if (pos > maxHeight) pos = maxHeight;
+m.style.top = pos + "px";
+if (pos === 0) {
+  m.classList.remove("up");
+  m.classList.add("down");
+}
+if (pos === maxHeight) {
+  m.classList.remove("down");
+  m.classList.add("up");
+}
+```
+
+In the optimized version, we first set the value of the `pos` variable by reading `m.style.top` instead of using `m.offsetTop`.  Using the element's inline style is faster because reading `m.offsetTop` forces the browser engine to know where all the elements are on the page, which requires the engine to recalculate the styles and the layout.
+
+We then set the new position of the icon like the previous version, but we don't read `m.offsetTop` again like we did before to adjust the icon's direction.
+
+The unoptimized code reads and writes the position of the icon from two different places, forcing the browser to recalculate the style and layout on each frame. The optimized version, however, writes and reads the position of the icon in the inline styles only.
+
+Note that this code could be made even faster by using CSS properties that only require the browser to do compositing, rather than layout.  Instead of manipulating the `top` property which forces the browser to run its layout code again, using the `transform` property would allow the browser to consider each icon as individual layers and display these layers in the right positions by re-compositing the final image.  For example, instead of using `m.style.top = pos + "px";`, we can use `m.style.transform = translateY(pos + "px,");`.
```

Comment:
```suggestion
Note that this code could be made even faster by using CSS properties that only require the browser to do compositing, rather than layout.  Instead of manipulating the `top` property, which forces the browser to run its layout code again, using the `transform` property would allow the browser to consider each icon as an individual layer, and then display these layers in the correct positions by re-compositing the final image.  For example, instead of using `m.style.top = pos + "px";`, we can use `m.style.transform = translateY(pos + "px,");`.
```
several wording changes for your consideration, as you see fit

Reply:
incorp'd